core/src/cluster_info.rs

```diff
@@ -1629,13 +1629,19 @@ impl ClusterInfo {
                 .flatten()
                 .collect()
         };
+
+        debug_assert!(pulls.len() < 2);
+        if pulls.len() > 1 {
+            error!("BUG: pulls.len() should be 0 or 1, not {}", pulls.len());
+        }
+
         self.append_entrypoint_to_pulls(thread_pool, &mut pulls);
         self.stats
             .new_pull_requests_count
             .add_relaxed(pulls.len() as u64);
         // There are at most 2 unique peers here: The randomly
         // selected pull peer, and possibly also the entrypoint.
-        let peers: Vec<Pubkey> = pulls.iter().map(|(peer, _, _, _)| *peer).dedup().collect();
+        let peers: Vec<Pubkey> = pulls.iter().map(|(peer, _, _, _)| *peer).collect();
```

Comment:
The above `debug_assert!()` helps point out how this `.dedup()` is not required.  `append_entrypoint_to_pulls()` only adds the entrypoint to `pulls` if it's not in the current CRDS table, which means that the entrypoint will never be in `pulls` twice.  So there's nothing to dedup

Reply:
Got it, thank you!